src/text_filter.cpp

```diff
@@ -0,0 +1,183 @@
+/**
+ * \brief This program converts the output files to meaningful files
+ *
+ * This program has a function output_filter which takes input and output
+ * files as arguments, it then read the input file line by line
+ * and converts it in to tabular formats. this function is called in the
+ * main cpp file of receiver,sender,subnet and top_model.
+ *
+ */
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "../include/filter.hpp"
+
+/**
+ * This function is used to trim the tailing space in the input
+ * @param ip of char
+ * @return ip
+ */
+
+char* trim(char* ip){
+	int c;
+	int index;
+	index = -1;
+	c = 0;
+	while(ip[c] != '\0'){
+	    if(ip[c] != ' ' && ip[c] != '\t' && ip[c] != '\n'){
+		    index = c;
+		}
+		c++;
+	}
+	ip[index+1] = '\0';
+	return ip;
+
+}
+
+/**
+ * This function is for arranging the unrecognizable output
+ * in to a correct readable format.
+ * @param input[]
+ * @param output[]
+ */
+
+void output_filter(const char *input,const char *output){
+
+	/**<This variables are used in various for loops for iteration*/
+	int i,j,k,l;
+
+	/**<This character array is used to store line by line data from input file*/
+	char data[300];
+
+	/**<This character arrays are used to store various results to write in output file*/
+	char component[300], port[300], time[300], port_value[300];
+
+	/**
+	 * Opening file pointers in read and write mode
+	 */
+
+	FILE *op = fopen(output,"w");
+	FILE *fp = fopen(input,"r");
+
+
+	/**
+	 * The below pointer to character is used to do various
+	 * operations on the file to get a pattern
+	 */
+	char *generated_by;
+	char *part1;
+	char *for_port;
+	char *value;
+
+
+	
+	/**
+	 * checking if there is any error in opening the out put file
+	 */
+	if(op == NULL){
+		printf("ERROR in opening file");
+	}
+
+	char first_line[500] = "TIME\t\t\tP_VALUE\t\t\tPORT\t\t\tCOMPONENT\n";
```

Comment:
please change "P_VALUE" to VALUE as per the instruction for task D,  Please check if you could orient the header of the file aligned with the contents so that the formatted output looks visually appealing.